testsuite/smoke-test/src/storage.rs

```diff
@@ -531,23 +533,30 @@ async fn test_db_restart() {
     for round in 0..3 {
         info!("{LINE} Restart round {round}");
         for (v, vid) in restarting_validator_ids.iter().enumerate() {
-            info!("{LINE} Round {round}: Restarting validator {v}.");
-            info!(
-                "{LINE} ledger info: {:?}",
-                client.get_ledger_information().await.unwrap(),
-            );
             let validator = swarm.validator_mut(*vid).unwrap();
             // sometimes trigger reconfig right before the restart, to expose edge cases around
             // epoch change
-            if rand::random::<usize>() % 3 == 0 {
-                info!("{LINE} Triggering reconfig right before restarting.");
+            if rand::random::<usize>() % 2 == 0 {
+                info!(
+                    "{LINE} Triggering reconfig right before restarting. Root account seq_num: {}. Ledger info: {:?}",
+                    pub_chain_info.root_account().sequence_number(),
+                    client.get_ledger_information().await.unwrap(),
+                );
                 reconfig(
-                    &validator.rest_client(),
```

Comment:
this is unfortunate.. I sent txns to the validator that's gonna be restarted right away and that's why sometimes some txns just disappear.

Reply:
hmm.. actually it shouldn't cause the failure because `reconfig()` waits for the reconfig to finish before proceeding..